packages/react/src/InlineMessage/InlineMessage.tsx

```diff
@@ -1,78 +1,104 @@
+    align-items: start;
+    color: var(--inline-message-fgColor, ${get('colors.fg.muted')});
+    line-height: var(--inline-message-lineHeight);
+    font-size: var(--inline-message-fontSize, ${get('fontSizes.1')});
 
-  &[data-variant='warning'] {
-    --inline-message-fgColor: ${get('colors.attention.fg')};
-  }
+    &[data-size='small'] {
+      --inline-message-fontSize: var(--text-body-size-small, ${get('fontSizes.0')});
+      --inline-message-lineHeight: var(--text-body-lineHeight-small, 1.6666);
+    }
 
-  &[data-variant='critical'] {
-    --inline-message-fgColor: ${get('colors.danger.fg')};
-  }
+    &[data-size='medium'] {
+      --inline-message-fontSize: var(--text-body-size-medium, ${get('fontSizes.1')});
+      --inline-message-lineHeight: var(--text-body-lineHeight-medium, 1.4285);
+    }
 
-  &[data-variant='success'] {
-    --inline-message-fgColor: ${get('colors.success.fg')};
-  }
+    &[data-variant='warning'] {
+      --inline-message-fgColor: ${get('colors.attention.fg')};
+    }
 
-  &[data-variant='unavailable'] {
-    --inline-message-fgColor: ${get('colors.fg.muted')};
-  }
+    &[data-variant='critical'] {
+      --inline-message-fgColor: ${get('colors.danger.fg')};
+    }
 
-  & .InlineMessageIcon {
-    min-height: calc(var(--inline-message-lineHeight) * var(--inline-message-fontSize));
-  }
-`
+    &[data-variant='success'] {
+      --inline-message-fgColor: ${get('colors.success.fg')};
+    }
+
+    &[data-variant='unavailable'] {
+      --inline-message-fgColor: ${get('colors.fg.muted')};
+    }
+
+    & .InlineMessageIcon {
+      min-height: calc(var(--inline-message-lineHeight) * var(--inline-message-fontSize));
+    }
+  `,
+)
 
 const variantToIcon: Record<MessageVariant, React.ReactNode> = {
-  warning: <AlertIcon className="InlineMessageIcon" />,
-  critical: <AlertIcon className="InlineMessageIcon" />,
-  success: <CheckCircleIcon className="InlineMessageIcon" />,
-  unavailable: <AlertIcon className="InlineMessageIcon" />,
+  warning: <AlertIcon className={classes.InlineMessageIcon} />,
+  critical: <AlertIcon className={classes.InlineMessageIcon} />,
+  success: <CheckCircleIcon className={classes.InlineMessageIcon} />,
+  unavailable: <AlertIcon className={classes.InlineMessageIcon} />,
 }
 
 const variantToSmallIcon: Record<MessageVariant, React.ReactNode> = {
-  warning: <AlertFillIcon className="InlineMessageIcon" size={12} />,
-  critical: <AlertFillIcon className="InlineMessageIcon" size={12} />,
-  success: <CheckCircleFillIcon className="InlineMessageIcon" size={12} />,
-  unavailable: <AlertFillIcon className="InlineMessageIcon" size={12} />,
+  warning: <AlertFillIcon className={classes.InlineMessageIcon} size={12} />,
+  critical: <AlertFillIcon className={classes.InlineMessageIcon} size={12} />,
+  success: <CheckCircleFillIcon className={classes.InlineMessageIcon} size={12} />,
+  unavailable: <AlertFillIcon className={classes.InlineMessageIcon} size={12} />,
```

Comment:
Should these also be behind a flag? Since the Styled component version will be looking for exactly "InlineMessageIcon" but with CSS Modules it will be hashed

Another option in the interim is that we could use the `.InlineMessageIcon` in CSS Modules so that these don't have to be flagged and we just use that class name in both flagged/unflagged scenarios

Reply:
yes, great callout!

Reply:
I'm not sure how that even worked before, pretty sure I tested on Storybook before I pushed these changes up